src/coreclr/jit/morph.cpp

```diff
@@ -13177,6 +13177,7 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                         GenTree* newOp2 = gtNewIconNode(-constVal, op1op2->TypeGet()); // -C
```

Comment:
I think we also want to update the VN for the new constant (this is what we do in other places). But since we haven't actually found a bug with it (yet), and the way to do it is kinda verbose right now, and I will be PRing changes that will make it less verbose soon, and the fact that this PR is likely to be backported, I think it is fine to leave this as is.

Reply:
Right, this ends up as a constant without any VN, but my impression was that we handle this ok downstream.

Reply:
I similarly haven't seen places in the optimizer that would trip up on a `NoVN` constant tree. I would not be surprised if they exist though, these contracts on what should and should not be maintained are very implicit.